codec-http2/src/main/java/io/netty/handler/codec/http2/DefaultHttp2RemoteFlowController.java

```diff
@@ -495,20 +518,22 @@ private void incrementPendingBytes(int numBytes) {
              * Note: this does not flush the {@link ChannelHandlerContext}.
              */
             int write(int allowedBytes) {
-                int before = payload.size();
-                needFlush |= payload.write(Math.max(0, allowedBytes));
-                int writtenBytes = before - payload.size();
                 try {
+                    pendingWriteQueue.remove();
```

Comment:
Yeah I think adding the method here is useful (I'm fine with either `complete` or `writeComplete`, no real preference).  It's simpler and clearer than popping the queue, which might even have unforeseen side-effects.  The documentation on the `write` method is very important here to make sure that we don't get on the Queue to Hell (Angus Young, the Netty project now owns that copyright! :)).